gobblin-api/src/main/java/gobblin/configuration/SourceState.java

```diff
@@ -92,8 +93,12 @@ public SourceState(State properties, List<WorkUnitState> previousTaskStates) {
   public synchronized Extract createExtract(Extract.TableType type, String namespace, String table) {
     Extract extract = new Extract(this, type, namespace, table);
     while (extractSet.contains(extract)) {
```

Comment:
Seems `extractSet` is not used anywhere. Why is this needed?
Also this method will be deprecated in #214 since properties in `SourceState` do not need to be added to `Extract`.